### PR TITLE
run: add a clarification about --always-changed and "callback" DVC-files

### PR DESCRIPTION
**Pull request by efiop**
Fixes https://github.com/iterative/dvc/issues/2843

Disregard the recommendations below if you use **Edit on Github** button to improve the docs in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This enables GitHub to link the PR to the corresponding bug and close it automatically when PR is merged.

Thank you for the contribution - we'll try to review and merge it as soon as possible. 🙏

### CODE REVIEW

**Comment by shcheklein** [comment]
@jorgeorpinel looks good? merging? It's been a while since it's created ...

**Comment by shcheklein** [comment]
@jorgeorpinel it's been a while. Could you please, take a look?

**Comment by jorgeorpinel** [comment]
Looking now.

File: static/docs/command-reference/run.md [hunk 1]
```diff
@@ -126,9 +126,11 @@ try creating a pipeline.
 - `--ignore-build-cache` - if an exactly equal DVC-file exists (same list of
   outputs and inputs, the same command to run), which has been already executed
   and is up to date, `dvc run` won't normally execute the command again (thus
-  "build cache"). This option gives a way to forcefully execute the command
-  anyway. It's useful if the command is non-deterministic (meaning it produces
-  different outputs from the same list of inputs).
+  "build cache"), unless it is an `--always-changed` DVC-file or it doesn't have
+  any dependencies (same as `--always-changed`). This option gives a way to
```
Comment (jorgeorpinel) [inline static/docs/command-reference/run.md]: This sentence was already too long and kind of confusing, this extra phrase really pushes it over the edge. Thanks for the update though. I'll fix it.

Comment (jorgeorpinel) [reply static/docs/command-reference/run.md]: Fixed in 75c6ea54.

File: static/docs/command-reference/run.md [hunk 2]
```diff
@@ -123,12 +123,14 @@ try creating a pipeline.
...
-  outputs and inputs, the same command to run), which has been already executed
-  and is up to date, `dvc run` won't normally execute the command again (thus
-  "build cache"). This option gives a way to forcefully execute the command
-  anyway. It's useful if the command is non-deterministic (meaning it produces
-  different outputs from the same list of inputs).
+- `--ignore-build-cache` - This options has an effect if an equivalent stage
+  file exists (same dependencies, outputs, and command to execute) which has
+  been already executed and is up to date. In this case, `dvc run` won't
+  normally execute the command again. The exception is when the existing stage
+  was is considered always changed (see `--always-changed` option). This option
```
Comment (jorgeorpinel) [inline static/docs/command-reference/run.md]: @efiop please double check correctness.

**Comment by jorgeorpinel** [comment]
I edited this so won't merge it myself and Ruslan may have some feedback, not sure. But feel free to merge otherwise. Thanks

File: static/docs/command-reference/run.md (hunk 2 above)
Comment (efiop) [reply static/docs/command-reference/run.md]: `was is considered` looks like a typo

Comment (efiop) [reply static/docs/command-reference/run.md]: Otherwise looks good.

Comment (jorgeorpinel) [reply static/docs/command-reference/run.md]: Thanks! Fixed in e4a73d17.

**Pull request merged**